### ISSUE TITLE
Preparation of Training Data with Label Embeddings .

### ISSUE THREAD

**Issue by NitinAggarwal1** [Extreme-classification/deepxml#16]
Hi , as per the documentation this is mentioned as way to create the training data set. 

DatasetName (e.g. LF-AmazonTitles-131K)
│   trn_X.txt   (text for trn documents, one text in each line)
|   tst_X.tst   (text for tst documents, one text in each line)
|   Y.txt       (text for labels, one text in each line)
│   trn_X_Y.txt (trn labels in spmat format)
|   tst_X_Y.txt (tst labels in spmat format)
|   filter_labels_test.txt (filter labels where label and test documents are same)
│
└───XXCondensedData (embeddings for tst, trn documents and labels, for benchmark datasets, XX=DX[Astec])
    │   trn_point_embs.npy (2D numpy matrix for trn document embeddings)
    │   tst_point_embs.npy (2D numpy matrix for tst document embeddings)
    |   label_embs.npy     (2D numpy matrix for label embeddings)

I could not understand the trn labels in spmat format . **Is there a script that creates that from input documents like ( trn_X.txt and tst_X.txt and Y.txt ) . This is for the case we want to use the label embeddings as well.** 

I want to generate it for my custom dataset.

**Comment by kunaldahiya** [comment]
Hi

In XC datasets, we just store the non-zero or relevant labels for each document as the number of labels can be really large and most of labels are irrelevant for a given document. So, we use sparse matrix (scipy.sparse.csr_matrix in particular) to handle our ground truth and predictions. trn_X_Y.txt file is just a csr_matrix dumped in sparse or libsvm format. Please use [this function](https://github.com/kunaldahiya/pyxclib/blob/985421a30a80380e172ca145399eae58134ae3f3/xclib/data/data_utils.py#L81) to create a txt file given a sparse matrix. [Here](https://github.com/kunaldahiya/pyxclib/tree/master#data-readingwriting) is an example. 

Moreover, if your dataset has label-text then I suggest [the NGAME algorithm](https://github.com/Extreme-classification/ngame) which is based on a transformer architecture and can make effective use of label-text.

PS: If you have your labels as list of lists, then [this function](https://github.com/kunaldahiya/pyxclib/blob/985421a30a80380e172ca145399eae58134ae3f3/xclib/utils/sparse.py#L205) can return csr_matrix. Let me know if something is unclear.

**Comment by NitinAggarwal1** [comment]
Thanks for the reply . I was under impression that DeepXML Algorithms are also using Label Text . I think this answer is complete for understanding how to use the DeepXML on custom training data.  I have some query on NGAME train data , I will add a question for that separately.
